lib/src/conflicts.rs

```diff
@@ -60,6 +60,27 @@ impl<T> Conflict<T> {
     pub fn set_add(&mut self, i: usize, value: T) {
         self.adds[i] = value;
     }
+
+    /// Remove pairs of entries that match in the removes and adds.
+    pub fn simplify(mut self) -> Self
+    where
+        T: PartialEq,
+    {
+        let mut add_index = 0;
+        while add_index < self.adds.len() {
+            let add = &self.adds[add_index];
+            add_index += 1;
+            for (remove_index, remove) in self.removes.iter().enumerate() {
+                if remove == add {
+                    self.removes.remove(remove_index);
+                    add_index -= 1;
+                    self.adds.remove(add_index);
+                    break;
+                }
+            }
+        }
+        self
+    }
 }
 
 impl Conflict<Option<TreeValue>> {
```

Comment:
I randomly noticed a typo on line 107 in the docstring to `to_backend_conflict`: "Note that the conversion is lossy" instead of "Note that the conflict is lossy"

Reply:
PR #1664 seems like a good place to fix that, so I'll do it there